### PR TITLE
@zephraph => Hide follows < 50, and immediately update follow counts

### PR DESCRIPTION
**Pull request by mzikherman**
Had to do some Relay stuff to update the store in `updater`, in addition to `optimisticResponse`.

The reason seems to be caching, I was still sometimes getting the same value back from Metaphysics/Gravity for the follow count, when following/unfollowing. I believe we should only need the `optimisticResponse` if everything worked, but without also manually setting it in `updater`, we would see the number change as soon as you pressed the button...and then it would flip back when the mutation actually finished.

### CODE REVIEW

File: src/Components/FollowButton/FollowArtistButton.tsx [hunk 1]
```diff
@@ -56,31 +57,42 @@ export class FollowArtistButton extends React.Component<Props> {
   handleFollow = () => {
     const { artist, currentUser, relay, onOpenAuthModal } = this.props
 
+    const newFollowCount = artist.is_followed
```
Comment (mzikherman) [inline src/Components/FollowButton/FollowArtistButton.tsx]: If you currently follow the artist, that means you have pressed the button and will now unfollow, so the new count is decremented.

File: src/Components/FollowButton/FollowArtistButton.tsx [hunk 2]
```diff
@@ -56,31 +57,42 @@ export class FollowArtistButton extends React.Component<Props> {
...
-            unfollow: artist.is_followed,
-          },
+          input: { artist_id: artist.id, unfollow: artist.is_followed },
         },
         optimisticResponse: {
           followArtist: {
             artist: {
               __id: artist.__id,
               is_followed: !artist.is_followed,
+              counts: { follows: newFollowCount },
```
Comment (mzikherman) [inline src/Components/FollowButton/FollowArtistButton.tsx]: This is enough to see the number in the header change immediately upon pressing the button!

File: src/Components/FollowButton/FollowArtistButton.tsx [hunk 3]
```diff
@@ -56,31 +57,42 @@ export class FollowArtistButton extends React.Component<Props> {
...
             },
           },
         },
+        updater: (store: RecordSourceSelectorProxy, data: SelectorData) => {
```
Comment (mzikherman) [inline src/Components/FollowButton/FollowArtistButton.tsx]: But, a second later it flashes back as we receive an incorrect value from Metaphysics. So let's just override that and just set it directly (we shouldn't _have_ to do this if that downstream caching was working).

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 71436a16f05e5ff6e3fa829c70a37f36b150e92e

https://deploy-preview-963--artsy-reaction.netlify.com

**Pull request merged**